qiskit_ibm_runtime/noise_learner/noise_learner.py

```diff
@@ -0,0 +1,251 @@
+                result_decoder=DEFAULT_DECODERS.get(self._program_id()),
+            )
+
+        if self._backend:
+            runtime_options["backend"] = self._backend
+            if "instance" not in runtime_options and isinstance(self._backend, IBMBackend):
+                runtime_options["instance"] = self._backend._instance
+
+        if isinstance(self._service, QiskitRuntimeService):
+            return self._service.run(
+                program_id=self._program_id(),
+                options=runtime_options,
+                inputs=inputs,
+                callback=options_dict.get("environment", {}).get("callback", None),
+                result_decoder=DEFAULT_DECODERS.get(self._program_id()),
+            )
+
+        return self._service.run(  # type: ignore[attr-defined]
+            program_id=self._program_id(),  # type: ignore[arg-type]
+            options=runtime_options,
+            inputs=inputs,
+        )
+
+    @classmethod
+    def _program_id(cls) -> str:
+        """Return the program ID."""
+        return "noise-learner"
+
+    def _set_options(
+        self, options: Optional[Union[Dict, NoiseLearnerOptions, EstimatorOptions]] = None
+    ) -> None:
+        """
+        Sets the options, ensuring that they are of type ``NoiseLearnerOptions``.
+        """
+        if not options:
+            self._options = NoiseLearnerOptions()
+        elif isinstance(options, NoiseLearnerOptions):
+            self._options = options
+        elif isinstance(options, EstimatorOptions):
+            d = asdict(options.resilience.layer_noise_learning)  # type: ignore[union-attr]
+            d.update({"twirling_strategy": options.twirling.strategy})  # type: ignore[union-attr]
+            d.update({"max_execution_time": options.max_execution_time})
+            d.update({"simulator": options.simulator})
+            d.update({"environment": options.environment})
+            d.update({"experimental": options.experimental})
+            self._options = NoiseLearnerOptions(**d)
+        else:
+            self._options = NoiseLearnerOptions(**options)
+
+    @staticmethod
+    def _get_inputs_options(options_dict: dict[str, Any]) -> dict[str, str]:
+        """Returns a dictionary of options that must be included in the program inputs,
+        filtering out every option that is not part of the NoiseLearningOptions."""
+        ret = {}
+
+        ignored_names = [
+            "_VERSION",
+            "max_execution_time",
+            "environment",
+        ]
+
+        for field in fields(NoiseLearnerOptions):
+            name = field.name
+            if name in options_dict and name not in ignored_names:
+                ret[name] = deepcopy(options_dict[name])
+
+        remove_dict_unset_values(ret)
+        remove_empty_dict(ret)
+
+        # Remove image
+        ret.get("experimental", {}).pop("image", None)
+
+        return ret
```

Comment:
we should add `support_qiskit` now, even if it's not used. So it'll be there when we are ready to support non-Qiskit callers.

Reply:
Okay, I added it. For now, the server will just `pop` it and forget about it